### ISSUE TITLE
Add comment to clarify Python2 vs. Python3 differences with InstanceType() and __new__()

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#8452]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)| @rodrigc reported|
|-|-|
|Trac ID|trac#8452|
|Type|enhancement|
|Created|2016-06-12 04:37:00Z|
|Branch|https://github.com/twisted/twisted/tree/8452-rodrigc-types|

In twisted/logger/_json.py , it would be nice to add a comment
to clarify differences in Python2 and Python3 with respect
to the availability of InstanceType() and __new__()

<details><summary>Searchable metadata</summary>

```
trac-id__8452 8452
type__enhancement enhancement
reporter__rodrigc rodrigc
priority__normal normal
milestone__Python_3_x Python-3.x
branch__8452_rodrigc_types 8452-rodrigc-types
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__None None
time__1465706220564322 1465706220564322
changetime__1466088189463266 1466088189463266
version__None None
owner__rodrigc rodrigc

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)<a name="note_1"></a>|@rodrigc commented|
|-|-|

See:
https://github.com/twisted/twisted/pull/172

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_2"></a>|@adiroiban set owner to @rodrigc|
|-|-|

changes look good. thanks for the cleanup. will merge.

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_3"></a>|@adiroiban set status to `closed`|
|-|-|

In changeset 20087d678ff1709fe969dd76bee0188decea1411
```
#!CommitTicketReference repository="" revision="20087d678ff1709fe969dd76bee0188decea1411"
Merge 8452-rodrigc-types: Clarify py2/py3 availability of InstanceType() and __new__().

Author: rodrigc
Reviewer: adiroiban
Fixes: #8452
```
